pkg/cloud/services/ec2/bastion.go

```diff
@@ -36,75 +36,78 @@ const (
 
 // ReconcileBastion ensures a bastion is created for the cluster
 func (s *Service) ReconcileBastion() error {
-	if s.scope.VPC().IsUnmanaged(s.scope.Name()) {
-		s.scope.V(4).Info("Skipping bastion reconcile in unmanaged mode")
-		return nil
-	}
+	if s.scope.AWSCluster.Spec.Bastion.Enabled {
+		if s.scope.VPC().IsUnmanaged(s.scope.Name()) {
+			s.scope.V(4).Info("Skipping bastion reconcile in unmanaged mode")
+			return nil
+		}
 
-	if !s.scope.AWSCluster.Spec.Bastion.Enabled {
-		if s.scope.AWSCluster.Status.Bastion != nil {
-			return s.DeleteBastion()
+		if !s.scope.AWSCluster.Spec.Bastion.Enabled {
+			if s.scope.AWSCluster.Status.Bastion != nil {
```

Comment:
Rather than using the existing Status here, this should use the output of `s.describeBastionInstance()` to determine if we should delete a previously created bastion instance.